packages/edit-site/src/components/post-edit-header/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Icon,
+	__experimentalText as Text,
+} from '@wordpress/components';
+import { useMemo } from '@wordpress/element';
+import {
+	privateApis as editorPrivateApis,
+	store as editorStore,
+} from '@wordpress/editor';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { sprintf, __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { PostCardPanel } = unlock( editorPrivateApis );
+
+export default function PostEditHeader( { postType, postId } ) {
+	const ids = useMemo( () => postId.split( ',' ), [ postId ] );
+	const { icon, labels } = useSelect(
+		( select ) => {
+			const { getEditedEntityRecord, getPostType } = select( coreStore );
+			const _record = getEditedEntityRecord(
+				'postType',
+				postType,
+				ids[ 0 ]
+			);
+
+			return {
+				icon: unlock( select( editorStore ) ).getPostIcon( postType, {
+					area: _record?.area,
+				} ),
+				labels: getPostType( postType )?.labels,
+			};
+		},
+		[ ids, postType ]
+	);
+
+	if ( ids.length === 1 ) {
+		return <PostCardPanel postType={ postType } postId={ ids[ 0 ] } />;
+	}
+
+	return (
+		<VStack spacing={ 1 } className="edit-site-post-edit-header">
+			<HStack spacing={ 2 } align="flex-start" justify="normal">
+				<Icon
+					className="edit-site-post-edit-header__icon"
+					icon={ icon }
+				/>
+				<Text
+					numberOfLines={ 2 }
+					truncate
+					className="edit-site-post-edit-header__title"
+					weight={ 500 }
+					as="h2"
+					lineHeight="20px"
```

Comment:
Ideally we use [variables](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-menu--with-submenu) or [mixins](https://github.com/WordPress/gutenberg/blob/5efeef9ae740d2aa51fa2493782773083b5363ee/packages/base-styles/_mixins.scss#L18-L70) for text styles rather than hard-coded values. Do you think it's feasible to apply `heading-medium()`?

Reply:
Sounds good, I updated that as part of: https://github.com/WordPress/gutenberg/pull/67390/commits/f35707b8a2aedb014fa1cb87f25027a27a2ffee0
I did have to update the mixins though, as they were relying on a line height variable that didn't exist, let me know if I updated it correctly: https://github.com/WordPress/gutenberg/pull/67390/files#diff-96351000444fb28f5c80a41909d185066c8202d23cdb19c69fc21f79f76960e8L18-R64

Also I did mimic the above from the `PostCardPanel` component https://github.com/WordPress/gutenberg/blob/f7a4f3f054e99e204b76393293fc4775b9596313/packages/editor/src/components/post-card-panel/index.js#L83 would it make sense to update that header to use the `heading-medium()` as well?

Reply:
> as they were relying on a line height variable that didn't exist

Oh good catch :D Thanks for fixing that. I guess this is the first time we're using the mixins :)

> would it make sense to update that header to use the heading-medium() as well?

Yes I think they should be the same.